Improve the interesting label troubleshooting section of CD doc (#7686)
jenkins-infra/jenkins.io: content/doc/developer/publishing/releasing-cd.adoc

== Troubleshooting

Check that `MAVEN_TOKEN` and `MAVEN_USERNAME` appear under Repository secrets.

> === No release has been triggered after merging a pull request

Author: Can these labels be added afterward then rerun the CD action? Asking as I'm not sure.

Reviewer: they can be yes

Reviewer 2: That may cause Release Drafter to rearrange release notes, but if you manually trigger the CD action it will publish regardless of any labels.

Reviewer 2: Or if you add an interesting label to a PR that was already merged, and then merge _another_ PR (even with an uninteresting label), then you will get a release, since the workflow will see that there has been at least interesting PR since the last release.

Reviewer 2: (It would be great if someone could volunteer to handle https://github.com/jenkins-infra/interesting-category-action/issues/1 which is probably not a lot of work, just picking an existing action and testing it on https://github.com/jenkinsci/jenkins-infra-test-plugin to make sure it behaves as expected.)

Author: Thanks for these complements, added in https://github.com/jenkins-infra/jenkins.io/pull/7686/commits/dcbc3ac887ae5b34d1cf27ead9bd78d9e42ef94d

I'll take a look at https://github.com/jenkins-infra/interesting-category-action/issues/1.

Ensure that the pull request has a label indicating it was of interest to users (e.g., `enhancement` rather than `chore`).

> See the complete list of label categories in https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml.

Reviewer: might be worth mentioning what is interesting? as developer is interesting but may not be known as such?

Author: Added the link to the interesting categories in https://github.com/jenkins-infra/jenkins.io/pull/7686/commits/847dd874e812fbd542e3665ea5f67f68fdbc01d9

> See the complete list of label categories in https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml.

Author: ```suggestion
See the complete list of label categories at https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml and the list of "interesting" categories at https://github.com/jenkins-infra/interesting-category-action/blob/main/action.yaml#L13.
```

> === The upload to the Maven repository fails with "401 Unauthorized"

Author: ```suggestion
If you add an "interesting" label on a pull request already merged then merge another pull request (even with an uninteresting label), you will get a release since the workflow will see that there has been at least one interesting pull request since the last release.

Finally, if you manually trigger the CD action it will publish regardless of any labels. (May cause cause Release Drafter to rearrange release notes)

=== The upload to the Maven repository fails with "401 Unauthorized"
```

== Troubleshooting

Check that `MAVEN_TOKEN` and `MAVEN_USERNAME` appear under Repository secrets.

=== No release has been triggered after merging a pull request

Ensure that the pull request has a label indicating it was of interest to users (e.g., `enhancement` rather than `chore`).

> See the complete list of label categories at https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml and the list of "interesting" categories at https://github.com/jenkins-infra/interesting-category-action/blob/main/action.yaml#L13.

Reviewer 2: I think it would be good to explicitly mention `developer` and explain that it should be used for a PR which does nothing on its own but is a dep of a downstream PR currently using an incremental version, so that the downstream can switch to a release version and be taken out of draft.

Author: Should be addressed in https://github.com/jenkins-infra/jenkins.io/pull/7686/commits/41d2f2099afd0ea1b11ea7e4dd5c448738a739b8

If you add an "interesting" label on a pull request already merged then merge another pull request (even with an uninteresting label), you will get a release since the workflow will see that there has been at least one interesting pull request since the last release.

> Finally, if you manually trigger the CD action it will publish a new release regardless of any labels.

Reviewer: This also depends on there being a passing build on the main/master branch (potentially worth capturing too)

Author: Something like that?
```suggestion
Finally, if you manually trigger the CD action and the build is passing on the primary branch, it will publish a new release regardless of any labels.
```

Reviewer: yeah

> Finally, if you manually trigger the CD action it will publish a new release regardless of any labels. (May cause cause Release Drafter to rearrange release notes)

Reviewer: unneeded, (also had a cause cause)
```suggestion
Finally, if you manually trigger the CD action it will publish a new release regardless of any labels.
```

== Troubleshooting

Check that `MAVEN_TOKEN` and `MAVEN_USERNAME` appear under Repository secrets.

=== No release has been triggered after merging a pull request

Ensure that the pull request has a label indicating it was of interest to users (e.g., `enhancement` rather than `chore`).

See the complete list of label categories at https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml and the list of "interesting" categories at https://github.com/jenkins-infra/interesting-category-action/blob/main/action.yaml#L13.

> Note about the `developer` category:  it should be used for a pull request which does nothing on its own but is a dep of a downstream pull request currently using an incremental version, so that the downstream can switch to a release version and be taken out of draft.

Reviewer: ```suggestion
You can use the developer label to release a pull request even if its not interesting to end users, e.g. it is a dependency of a downstream pull request or it fixes tests for Plugin Compatibility Testing.
```

Author: Thanks, much more clear!
